app/controllers/applicationController.js

```diff
@@ -74,7 +96,27 @@ updateApplication = async (req, res) => {
 };
 
 deleteApplication = async (req, res) => {
+    const id = req.params.id;
+
+    const application = await Application.findById(id);
+    if (!application) {
+        return res.status(201).json({
+            message: `Application ${id} not found`,
+        });
+    }
```

Comment:
Now since we're not handling any errors thrown by Application.findById(), when mongoose runs into a bad id it causes the server to crash. 

<img width="1131" alt="Screen Shot 2022-03-04 at 9 33 07 AM" src="https://user-images.githubusercontent.com/45221526/156784526-f7aa74c6-f632-4df0-a109-56fe216b11ef.png">

However, this also brought to my attention an issue with updateApplication where the api call will get hung up if the id is "valid" but the resource doesn't exist.